packages/rest/src/rest-http-error.ts

```diff
@@ -7,12 +7,22 @@ export namespace RestHttpErrors {
     extraProperties?: Props,
   ): HttpErrors.HttpError & Props {
     const msg = `Invalid data ${JSON.stringify(data)} for parameter ${name}!`;
-    return Object.assign(new HttpErrors.BadRequest(msg), extraProperties);
+    return Object.assign(
+      new HttpErrors.BadRequest(msg),
+      {
+        code: 'INVALID_PARAMETER_VALUE',
+        parameterName: name,
+      },
+      extraProperties,
+    );
   }
 
   export function missingRequired(name: string): HttpErrors.HttpError {
     const msg = `Required parameter ${name} is missing!`;
-    return new HttpErrors.BadRequest(msg);
+    return Object.assign(new HttpErrors.BadRequest(msg), {
+      code: 'MISSING_REQUIRED_PARAMETER',
```

Comment:
Can we add these new error codes to the mapping you created for the `404` case?

Reply:
These errors are created inside REST layer, where we know about HTTP status codes and thus `error.statusCode` is already set. No need to set `statusCode` in the `reject` action. In fact, the mapping preserves `statusCode` if it's already set, and uses the lookup table only when no status code was provided by the error. Adding these new error codes to the lookup table would be a no-op that may confuse users IMO.